scrapers_next/ma/committees.py

```diff
@@ -0,0 +1,100 @@
+from spatula import URL, HtmlListPage, HtmlPage, CSS, XPath, selectors
+from openstates.models import ScrapeCommittee
+import requests
+
+requests.packages.urllib3.disable_warnings()
+
+
+class CommitteeDetail(HtmlPage):
+    def process_page(self):
+        com = self.input
+        try:
+            for members_area in XPath(
+                '//*[contains(@class, "committeeMemberList")]'
+            ).match_one(self.root)[0]:
+                if "There are no members" in members_area.text:
+                    break
+                # continue to extract members if available
+                for members_info in XPath(
+                    '//*[contains(@class, "committeeMemberList")]'
+                ).match_one(members_area):
+                    for data in members_info:
+                        name = None
+                        role = "Member"
+                        for info in data.text_content().splitlines():
+                            if info.strip():
+                                if name is None:
+                                    name = info.strip()
+                                else:
+                                    role = info.strip()
+                        com.add_member(name, role)
+        except selectors.SelectorError:
+            return com
+        com.add_source(
+            self.source.url,
+            note="Committee Details Page",
+        )
+        return com
+
+
+class CommitteeList(HtmlListPage):
+    source = URL(
+        "https://malegislature.gov/Committees",
+        headers={
+            "User-Agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:108.0) Gecko/20100101 Firefox/108.0",
+            "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8",
+            "Accept-Language": "en-US,en;q=0.5",
+            "Referer": "https://www.google.com/",
+            "Connection": "keep-alive",
+            "Upgrade-Insecure-Requests": "1",
+            "Sec-Fetch-Dest": "document",
+            "Sec-Fetch-Mode": "navigate",
+            "Sec-Fetch-Site": "cross-site",
+            "Sec-Fetch-User": "?1",
+            "Pragma": "no-cache",
+            "Cache-Control": "no-cache",
+        },
+        verify=False,
+    )
+
+    def process_page(self):
+        for comm in XPath('//*[contains(@class, "committeeList")]').match(self.root):
+            for each_committee in comm:
+                if comm is not None:
+                    name = str(each_committee.text_content())
+                    if (
```

Comment:
How about `if any(name.startswith(s) for s in ["House", "Senate", "Joint"]):`? Makes it easier to expand/contract later if need be.

Reply:
actually I'm skipping temporary committees

Reply:
done :) thanks

Reply:
Sorry for giving you mixed signals with my request to remove all filters here! But it will benefit us to have the data for temporary and special committees also, so I appreciate you making the [requested change](https://github.com/openstates/openstates-scrapers/pull/4426/files#r1089225916).